File: CS-Examples/09_Charts/SetLegendBackgroundColor.cs
using System;
using System.Windows.Forms;
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet.Charts;
using System.Drawing;

namespace SetLegendBackgroundColor
{

	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, EventArgs e)
		{
            //Create a workbook
			Workbook workbook = new Workbook();

            //Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ChartSample1.xlsx");

            // Get the first worksheet
            Worksheet ws = workbook.Worksheets[0];

            // Get the chart from the worksheet
            Chart chart = ws.Charts[0];

            // Access the legend frame format and set the background color
            XlsChartFrameFormat x = chart.Legend.FrameFormat as XlsChartFrameFormat;
            x.Fill.FillType = ShapeFillType.SolidColor;
            x.ForeGroundColor = Color.SkyBlue;

            // Save the modified workbook
            string output = "SetLegendBackgroundColor.xlsx";
            workbook.SaveToFile(output, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the file
            ExcelDocViewer(output);
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: VB-Examples/09_Charts/SetLegendBackgroundColor.vb
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet.Charts

Namespace SetLegendBackgroundColor

	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new instance of Workbook
            Dim workbook As New Workbook()

            ' Load the Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ChartSample1.xlsx")

            ' Get the first worksheet from the workbook
            Dim ws As Worksheet = workbook.Worksheets(0)

            ' Get the first chart from the worksheet
            Dim chart As Chart = ws.Charts(0)

            ' Cast the Legend's FrameFormat to XlsChartFrameFormat
            Dim x As XlsChartFrameFormat = TryCast(chart.Legend.FrameFormat, XlsChartFrameFormat)

            ' Set the fill type of the legend frame to solid color
            x.Fill.FillType = ShapeFillType.SolidColor

            ' Set the foreground color (background color) of the legend frame to SkyBlue
            x.ForeGroundColor = Color.SkyBlue

            ' Specify the output file name for saving the modified workbook
            Dim output As String = "SetLegendBackgroundColor.xlsx"

            ' Save the workbook to the specified output file with Excel 2013 format
            workbook.SaveToFile(output, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the file
            ExcelDocViewer(output)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
